orchestrate-containers/collab.md

```diff
@@ -737,3 +932,148 @@ kubectl create -f deployWordPress.yml
+---
+- hosts: all
+  remote_user: root
+
+  tasks:
+    - name: install dependencies
+      apt:
+        name:
+            - apt-transport-https
+            - ca-certificates
+            - curl
+            - gnupg2
+            - software-properties-common
+            - python-pip
+            - python-apt
+            - openssl
+        state: present
+    - name: adding apt-key for docker
+      apt_key:
+        url: https://download.docker.com/linux/debian/gpg
+        state: present
+
+    - name: adding docker repo list
+      apt_repository:
+        repo: deb [arch=amd64] https://download.docker.com/linux/debian stretch stable
+        state: present
+
+    - name: install docker-ce and docker-compose
+      apt:
+        name:
+            - docker-compose
+            - docker-ce
+        state: present
+
+    - name: adding apt-key for kube
+      apt_key:
+        url: https://packages.cloud.google.com/apt/doc/apt-key.gpg
+        state: present
+
+    - name: adding kube repo list
+      apt_repository:
+        repo: deb http://apt.kubernetes.io/ kubernetes-xenial main
+        state: present
+
+    - name: install docker-ce and docker-compose
+      apt:
+        name:
+            - kubeadm
+            - kubectl
+            - kubelet
+        state: present
+
+    - name: Commenting a line using the regualr expressions in Ansible.
+      replace:
+        path: /etc/fstab
+        regexp: '(.*swap.*sw.*)'
+        replace: '#\1'
+
+    - name: Disable swap
+      command: swapoff -a
+      when: ansible_swaptotal_mb > 0
+```
+
+Create a script to set up the kubernetes cluster
+
+```bash
+---
+- hosts: all
+  remote_user: root
+
+  vars_prompt:
+    - name: "MasterIP"
+      prompt: "Please enter the Master Node IP address that has connection with the rest of the nodes in the cluster (e.g 192.168.0.100)"
+      default: "192.168.100.100"
+      private: no
+
+    - name: "SlaveName"
+      prompt: "Please enter the actual hostname of the Slave Node (e.g node1)"
+      default: "node1"
+      private: no
+
+    - name: "SlaveIP"
+      prompt: "Please enter the Slave Node IP address that resides in the same network range as the Master Node (e.g 192.168.100.101)"
+      default: "192.168.100.101"
+      private: no
+
+  tasks:
+    - name: Check if etcd file exists
+      stat: path=/etc/kubernetes/manifests/etcd.yaml
+      register: etcd
+
+    - name: Initialized the Master Node using canal as the pod network
+      shell: "kubeadm init --pod-network-cidr=10.244.0.0/16 --apiserver-advertise-address={{MasterIP}}"
+      when: etcd.stat.exists == False
+
+    - name: Check if Kubernetes admin config exists
+      stat: path=$HOME/.kube/config
+      register: kubeconfig
+
+    - name: Setup Master Node
+      shell: |
+        sysctl net.bridge.bridge-nf-call-iptables=1
+        mkdir -p $HOME/.kube
+        sudo cp /etc/kubernetes/admin.conf $HOME/.kube/config
+        sudo chown $(id -u):$(id -g) $HOME/.kube/config
```

Comment:
Using the `shell` or `command` modules is brute forcing your way thru the configuration. I understand that there's a lack of kubernetes setup modules but please avoid it where you can.

This example in particular is a bad use of `shell`. Never concat multiple commands together. Each Ansible task is supposed to be idempotent, remember? If I run this script 2 times, can you guarantee that the end result is the same? Please verify.

When using `shell` or `command`, try the `creates` argument to ensure a task's idempotency.

Reply:
For this particular segment, the task is to copy the admin configuration to the config file. Therefore it is okay for the command to be run again as it will just be overwriting the previous file. 

The script was created as a template for us so that we can improve on it. Which is why it contains a lot of `shell` commands, we will work on improving it.

Reply:
Ok, please improve it in the next version.